### PR TITLE
#2686 [Manager] Prevent connector from making an unneccesary put request when switching between connectors

### PR DESCRIPTION
**Pull request by konekoya**
### Checklist
- [x] [Review Contribution Guidelines](https://github.com/oharastream/ohara/blob/master/CONTRIBUTING.md)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#authors))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #2686)
- [ ] Add unit tests (or please explain why this PR is not worth having new tests)
- [ ] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-manager/client/src/components/pages/PipelinePage/Connectors/connectorUtils.js [hunk 1]
```diff
@@ -219,11 +219,11 @@ export const changeKeySeparator = key => {
 export const useFetchConnectors = props => {
   const [state, setState] = useState(null);
   const [configs, setConfigs] = useState(null);
+  const { group } = props.pipeline;
+  const { connectorName } = props.match.params;
```
Comment (konekoya) [inline ohara-manager/client/src/components/pages/PipelinePage/Connectors/connectorUtils.js]: Moving these object destructuring outsides of the `useffect` can prevent the unnecessary updates

File: ohara-manager/client/src/components/pages/PipelinePage/PipelineEditPage/PipelineEditPage.js [hunk 2]
```diff
@@ -84,13 +84,18 @@ const PipelineEditPage = props => {
...
+    // If the pipeline data is not ready and prevPipeline is equal to pipeline
+    // don't load the graph
+    if (isEmpty(pipeline) || isEqual(prevPipeline, pipeline)) return;
 
     // Pipeline data is ready, let's load the graph
     loadGraph(pipeline);
+  }, [loadGraph, pipeline, prevPipeline]);
+
+  useEffect(() => {
+    if (isEmpty(pipeline) || !isEmpty(connectors)) return;
```
Comment (konekoya) [inline ohara-manager/client/src/components/pages/PipelinePage/PipelineEditPage/PipelineEditPage.js]: Make sure we only want to fetch worker data when pipeline is not empty and connectors state is not set yet

File: ohara-manager/client/package.json [hunk 3]
```diff
@@ -24,7 +24,6 @@
   "dependencies": {
     "@material-ui/core": "4.3.3",
     "@material-ui/icons": "4.2.1",
-    "@material-ui/styles": "4.3.3",
```
Comment (konekoya) [inline ohara-manager/client/package.json]: This seems to be causing trouble sometimes, I'm removing since @material-ui/core already contains this package

**Comment by wu87988622** [comment]
retry manager-api

File: ohara-manager/client/src/components/pages/PipelinePage/Connectors/AutoSave.js [hunk 4]
```diff
@@ -36,10 +37,15 @@ class AutoSave extends React.Component {
...
   save = debounce(async () => {
-    const { values, save, updateHasChanges } = this.props;
+    const { values, save, updateHasChanges, connectorState } = this.props;
     const difference = diff(this.state.values, values);
 
-    if (difference && difference.length) {
+    // If the connector is running, we shouldn't update since it will cause
+    // server side error
+    const shouldUpdate =
+      difference && difference.length && isNull(connectorState);
```
Comment (konekoya) [inline ohara-manager/client/src/components/pages/PipelinePage/Connectors/AutoSave.js]: Temp workaround for now. I don't how to fix this issue. Will need more time to dig in

**Comment by konekoya** [comment]
@wu87988622 Hmm, it's all green locally. Not sure what happened, let's retry and see if it can pass this time

**Comment by konekoya** [comment]
retry manager-e2e

**Comment by konekoya** [comment]
retry manager-e2e

**Comment by konekoya** [comment]
retry manager-e2e

**Comment by konekoya** [comment]
retry manager-e2e

File: ohara-manager/scripts/runE2eTests.js [hunk 5]
```diff
@@ -100,7 +98,6 @@ const run = async (prod, apiRoot, serverPort = 5050, clientPort = 3000) => {
         stdio: 'inherit',
       },
     );
-    console.log('client.pid', client.pid);
```
Comment (konekoya) [inline ohara-manager/scripts/runE2eTests.js]: These console.logs are not that helpful

**Comment by konekoya** [comment]
retry manager-e2e

**Comment by konekoya** [comment]
@wu87988622 I'm running into a very odd issue. The stream app e2e test suite is failing constantly on Jenkins (CI mode) but passed in both CI and GUI modes locally. It would be create if you can help me figure out why it's failing... ❤️

**Comment by konekoya** [comment]
retry manager-e2e

**Comment by konekoya** [comment]
retry manager-e2e

**Comment by konekoya** [comment]
retry manager-e2e

**Comment by konekoya** [comment]
A lot of our E2E tests are failing right now. I have filed an issue to address this #2820

**Pull request merged**